Reporting/M5 Document/Reflection.md

```diff
@@ -0,0 +1,29 @@
+
+- In the future, we would ensure a more comprehensive review of security and messaging features during the requirement gathering phase to avoid oversights.
+
+## What did you miss in your initial planning for the project (beyond just the requirements)?
+- In our initial planning, we may have overlooked the potential impact of external dependencies and integration challenges. Specifically, we did not anticipate the complexities associated with integrating encryption protocols into our messaging system or the time required for thorough testing across different platforms and devices.
+
+- In the future, we would allocate more time and resources to address these aspects during the planning phase to minimize delays during development.
+
+## Would you (as a team) deal with testing differently in the future?
+
+- Looking back, we recognize that our testing approach could have been more comprehensive. While we conducted unit tests and integration tests throughout the development process, we underestimated the importance of user acceptance testing (UAT) and end-to-end testing.
+
+- In the future, we would prioritize UAT sessions with real users to gather feedback early and address any usability issues proactively.
+
+## If you were to estimate the efforts required for this project again, what would you consider?  (Really I am asking the team to reflect on the difference between what you thought it would take to complete the project vs what it actually took to deliver it). 
+
+- When estimating the efforts required for this project initially, we primarily considered the complexity of the features, the experience level of our team members, and the projected timeline. However, in hindsight, we underestimated the time required for debugging and troubleshooting, particularly when integrating third-party libraries and implementing encryption functionalities.
+
+- In the future, we would factor in a buffer for unforeseen challenges and prioritize incremental releases to facilitate early feedback and course correction.
+What did your team do that you feel is unique or something that the team is especially proud of (was there a big learning moment that the team had in terms of gaining knowledge of a new concept/process that was implemented).
+One unique aspect of our project was the implementation of encryption-oriented peer-to-peer messaging. This required a deep dive into encryption protocols and security best practices, which was a significant learning moment for our team. Additionally, we successfully integrated Next.js as the React framework for our front-end, leveraging its efficiency and performance benefits. Overall, our team is proud of our ability to tackle complex challenges and deliver a secure and user-friendly social platform.
```

Comment:
- We don't have encrypted messaging. We do, however, have encrypted channel keys, which means that the actual keys themselves are impossible to guess, so you can't get into someone's channel without being invited to one. The messages themselves are not encrypted. But yes, that still involves a deep dive into encryption protocols, especially security best practices which was completely overwhelming.
- Proud of the most: `draggable text channels, our home page, encrypted channel keys, our use of webhooks and a proxy server, our privacy toggle, real-time text channel creation across all parties, our friend request system, responsive design, UI design, use of a complicated but modern tech stack -- Next.js + MongoDB + TypeScript.